action/protocol/staking/protocol.go

```diff
@@ -650,40 +686,51 @@ func (p *Protocol) calculateVoteWeight(v *VoteBucket, selfStake bool) *big.Int {
 	return CalculateVoteWeight(p.config.VoteWeightCalConsts, v, selfStake)
 }
 
+type nonceUpdateType bool
+
+const (
+	updateNonce   nonceUpdateType = true
+	noUpdateNonce nonceUpdateType = false
+)
```

Comment:
feel there's no need for this, using bool is enough, the following logic should be clear
```
if nonceUpdated {
   // updated in createNFTBucket, no need to update again in settleAction
   nonceUpdated = false
}
```